src/main/java/dev/jbang/cli/BaseBuildCommand.java

```diff
@@ -170,7 +168,12 @@ public static IntegrationResult buildJar(ScriptSource src, RunContext ctx, File
 		Util.infoMsg("Building jar...");
 		Util.verboseMsg("compile: " + String.join(" ", optionList));
 
-		Process process = new ProcessBuilder(optionList).inheritIO().start();
+		final ProcessBuilder processBuilder = new ProcessBuilder(optionList).inheritIO();
+		if (src instanceof GroovyScriptSource) {
```

Comment:
why the need for changing JAVA_HOME ?

and then why remove GROOY_HOME?

Reply:
Code snippet from startGroovy(called by groovyc):  

```bash
# For Cygwin, ensure paths are in UNIX format before anything is touched.
if $cygwin ; then
    [ -n "$GROOVY_HOME" ] && GROOVY_HOME="$(cygpath --unix "$GROOVY_HOME")"
    [ -n "$JAVACMD" ] && JAVACMD="$(cygpath --unix "$JAVACMD")"
    [ -n "$JAVA_HOME" ] && JAVA_HOME="$(cygpath --unix "$JAVA_HOME")"
    [ -n "$CP" ] && CP=$(cygpath --path --unix "$CP")
else
    if [ -n "$GROOVY_HOME" ] && [ "$(expr "$GROOVY_HOME" : '\/$')" ] ; then
        GROOVY_HOME="$(echo "$GROOVY_HOME" | sed -e 's/\/$//')"
    fi
fi
```

If developer has defined JAVA_HOME to 8 and GROOVY_HOME to 2.5.6 already,  but the Groovy script use `//JAVA 17` and `//GROOVY 3.0.9`,  and groovyc could not compile groovy code to class correctly because groovyc will use env variables.  For examples:

* Script uses API from Java 17 only
* Script uses API from Groovy 3.0 only

So the solution is to reset JAVA_HOME and remove GROOVY_HOME.  It's ok without GROOVY_HOME env, and groovyc will use its parent directory as GROOVY_HOME.

Reply:
Just that we do set java home in some cases I'm not fond of the imbalance. But since this code at the moment only affect groovy I merged it and we iterate on it.